### ISSUE TITLE
Sklearn model error (TypeError: fit() takes 3 positional arguments but 4 were given)

### ISSUE THREAD

**Issue by laylagerami** [deepchem/deepchem#3085]
## 🐛 Bug

When using certain Sklearn models (specifically GaussianProcessClassifier, KNeighborsClassifier, MLPClassifier and QuadraticDiscriminantAnalysis) the model.fit() stage throws an error.

## To Reproduce

1.  model = dc.models.SklearnModel(GaussianProcessClassifier())
2. model.fit(train_dataset)

```
TypeError Traceback (most recent call last)
/tmp/ipykernel_29886/[3380587234.py](http://3380587234.py/) in <cell line: 2>()

  1 model = dc.models.SklearnModel(GaussianProcessClassifier())
----> 2 model.fit(train_dataset)
3 model.predict(test_dataset)

~/anaconda3/envs/tensorflow2_p38/lib/python3.8/site-packages/deepchem/models/sklearn_models/sklearn_model.py in fit(self, dataset)
109 # Some scikit-learn models don't use weights.
110 if self.use_weights:
--> 111 self.model.fit(X, y, w)
112 return
113 self.model.fit(X, y)

TypeError: fit() takes 3 positional arguments but 4 were given
```
## Expected behavior

Expected to fit the model. When using e.g., model = dc.models.SklearnModel(RandomForestClassifier()) the model fits/predicts as expected. 

## Environment

* OS: Linux
* Python version: 3.8
* DeepChem version: [pre] installation (2.6.2.dev)

**Comment by zzzqiii** [comment]
I've encountered the same problem when tring KNeighborsClassifier model:

```
sklearn_model = KNeighborsClassifier()
model = dc.models.SklearnModel(sklearn_model)
model.fit(train_dataset)
```

`TypeError: validation_step() takes 3 positional arguments but 4 were given` 

Python version: 3.9.0
DeepChem version: 2.6.1

**Comment by zzzqiii** [comment]
I've encountered the same problem when tring KNeighborsClassifier model:

```
sklearn_model = KNeighborsClassifier()
model = dc.models.SklearnModel(sklearn_model)
model.fit(train_dataset)
```

`TypeError: validation_step() takes 3 positional arguments but 4 were given` 

Python version: 3.9.0
DeepChem version: 2.6.1

**Comment by arunppsg** [comment]
Sorry for the delayed response but the reason for this is that methods like KNeighborsClassifier, GaussianProcessClassifier does not accepts a weights argument in the their call to fit but deepchem assumes that they accept a weight argument and passes an additional argument, thereby the error `3 positional arguments but 4 were given` (weights as the forth one).

**Comment by arunppsg** [comment]
Fixed in #3101
